modules/host/network.go

```diff
@@ -18,14 +18,14 @@ var rpcSettingsDeprecated = types.Specifier{'S', 'e', 't', 't', 'i', 'n', 'g', '
 // checks if the host's hostname has changed, and makes an updated host
 // announcement if so.
 func (h *Host) threadedUpdateHostname() {
+	err := h.tg.Add()
+	if err != nil {
+		return
+	}
+	defer h.tg.Done()
+
 	for {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
 		h.managedLearnHostname()
-		h.tg.Done()
-
```

Comment:
no you probably could not. Depends on how many goroutines are running I would think.